pkg/reconciler/common/tektoninstallerset/install_test.go

```diff
@@ -0,0 +1,110 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tektoninstallerset
+
+import (
+	"context"
+	"path"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	mf "github.com/manifestival/manifestival"
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/client/clientset/versioned/fake"
+	"github.com/tektoncd/pipeline/test/diff"
+	"gotest.tools/v3/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCreateInstallerset(t *testing.T) {
+	testData := path.Join("testdata", "release.yaml")
+
+	manifest, err := mf.ManifestFrom(mf.Recursive(testData))
+	assert.NilError(t, err)
+
+	di := NewDefaultInstaller()
+	di.AddManifest(manifest)
+
+	di.AddLabelsFromMap(map[string]string{
+		v1alpha1.CreatedByKey: "pipeline",
+	})
+
+	di.AddAnnotationsFromMap(map[string]string{
+		"ns": "tekton-pipelines",
+	})
+
+	fakeClientset := fake.NewSimpleClientset()
+
+	client := fakeClientset.OperatorV1alpha1().TektonInstallerSets()
+
+	newISM := newTisMetaWithName("pipeline")
+
+	createdIs, err := createInstallerSetWithClient(context.Background(), client, di, newISM)
+	assert.Equal(t, err, nil)
+
+	labels := map[string]string{v1alpha1.CreatedByKey: "pipeline"}
+	annotations := map[string]string{"ns": "tekton-pipelines"}
+
+	expectedIs := &v1alpha1.TektonInstallerSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "pipeline",
+			Labels:          labels,
+			Annotations:     annotations,
+			OwnerReferences: []metav1.OwnerReference{{}},
+		},
+		Spec: v1alpha1.TektonInstallerSetSpec{
+			Manifests: manifest.Resources(),
+		},
+	}
+
+	if d := cmp.Diff(createdIs, expectedIs); d != "" {
+		t.Errorf("Actual created installerset is different from the expected one %s", diff.PrintWantGot(d))
+	}
+}
+
+func TestMakeInstallerset(t *testing.T) {
+	testData := path.Join("testdata", "release.yaml")
+
+	manifest, err := mf.ManifestFrom(mf.Recursive(testData))
+	assert.NilError(t, err)
+
+	tis := newTisMetaWithName("pipeline")
+
+	tis.Labels = map[string]string{v1alpha1.CreatedByKey: "pipeline"}
+	tis.Annotations = map[string]string{"ns": "tekton-pipelines"}
+
+	actual := makeInstallerSet(&manifest, tis)
+
+	labels := map[string]string{v1alpha1.CreatedByKey: "pipeline"}
+	annotations := map[string]string{"ns": "tekton-pipelines"}
+
+	expected := &v1alpha1.TektonInstallerSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "pipeline",
+			Labels:          labels,
+			Annotations:     annotations,
+			OwnerReferences: []metav1.OwnerReference{{}},
+		},
+		Spec: v1alpha1.TektonInstallerSetSpec{
+			Manifests: manifest.Resources(),
+		},
+	}
+
+	if d := cmp.Diff(actual, expected); d != "" {
+		t.Errorf("Actual installerset is different from the expected one %s", diff.PrintWantGot(d))
```

Comment:
let us write the PrintWantGot in tektoncd/operator instead of vendoring it from [tektoncd/pipeline PrintWantGot](https://github.com/tektoncd/pipeline/blob/0d8b26a35c2705eba7da98c3c014b6e5e80417cd/test/diff/print.go#L25).

It is a tiny 1 line funtion. 


```plaintext
A little copying is better than a little dependency.
- Go Proverb
```
🤓

Reply:
or else we can directly use the `fmt.Sprintf("(-want, +got): %s", diff)` format string in this test.